<?xml version="1.0"?>

<launch>

  <include file="$(find horti_moveit_config)/launch/demo.launch" pass_all_args="true">
    <!-- robot description is loaded by gazebo.launch, to enable Gazebo features -->
    <arg name="load_robot_description" value="false" />
    <arg name="moveit_controller_manager" value="ros_control" />
    <arg name="use_rviz" default="false"/>
    <arg name="pipeline" default="ompl" />
  </include>

  <include file="$(find horti_moveit_config)/launch/gazebo.launch" >
    <arg name="world_path" default="$(find salih_marangoz_thesis)/worlds" />
    <arg name="world_name" default="world14" />
    <arg name="paused" default="false"/>
    <arg name="gui" value="true"/>
  </include>

  <node name="gazebo_killer_node" pkg="salih_marangoz_thesis" type="gazebo_killer_node.py"/>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find salih_marangoz_thesis)/rviz/start.rviz"/>

  <node pkg="capsicum_superellipsoid_detector" type="simple_roi_detector_node" name="detect_roi_noisy">
    <remap from="~input" to="/head_camera_d435i/depth_registered/points" />
    <remap from="~roi_cloud" to="/detect_roi_noisy/roi_cloud" />
    <remap from="~nonroi_cloud" to="/detect_roi_noisy/nonroi_cloud" />
    <param name="map_frame" value="world" />Â´
  </node>

  <node pkg="topic_tools" type="throttle" name="roi_cloud_throttle" args="messages /detect_roi_noisy/roi_cloud 1.0 /detect_roi_noisy/roi_cloud_throttle"/>

  <node pkg="capsicum_superellipsoid_detector" type="detector_node" name="capsicum_superellipsoid_detector" output="screen" respawn="true">
    <remap from="pc_in" to="/detect_roi_noisy/roi_cloud_throttle"/>

    <param name="p_cost_type" value="2"/>
    <param name="p_prior_center" value="0.1"/>
    <param name="p_prior_scaling" value="0.1"/>
    <param name="p_missing_surfaces_num_samples" value="300"/>
    <param name="p_missing_surfaces_threshold" value="0.015"/>
    <param name="p_min_cluster_size" value="100"/>
    <param name="p_max_cluster_size" value="10000"/>
    <param name="p_max_num_iterations" value="100"/>
    <param name="p_cluster_tolerance" value="0.01"/>
    <param name="p_estimate_normals_search_radius" value="0.015"/>
    <param name="p_estimate_cluster_center_regularization" value="2.5"/>
    <param name="p_pointcloud_volume_resolution" value="0.001"/>
    <param name="p_octree_volume_resolution" value="0.001"/>
    <param name="p_print_ceres_summary" value="false"/>
    <param name="p_use_fibonacci_sphere_projection_sampling" value="false"/>
    <param name="p_world_frame" value="world"/>
  </node>



</launch>
